python/packages/core/agent_framework/_mcp.py

```diff
@@ -510,8 +510,17 @@ async def load_prompts(self) -> None:
                 exc_info=exc,
             )
             prompt_list = None
+        
+        # Track existing function names to prevent duplicates
+        existing_names = {func.name for func in self._functions}
```

Comment:
this is good, but in addition to this, I would also like to see logic that prevents multiple calls to `load_tools` and `load_prompts` because that also shouldn't happen by accident.